apps/opendosm/dashboards/services-producer-prices/index.tsx

```diff
@@ -0,0 +1,322 @@
+
+interface LabourProductivityOptions {
+  growth_qoq: Record<(typeof ServicesPPIData)[number], number>;
+  growth_yoy: Record<(typeof ServicesPPIData)[number], number>;
+  index: Record<(typeof ServicesPPIData)[number], number>;
+}
+
+interface ServicesPPIProp {
+  last_updated: string;
+  timeseries: WithData<LabourProductivityOptions>;
+  timeseries_callout: WithData<LabourProductivityOptions>;
+}
+
+const ServicesProducerPrices: FunctionComponent<ServicesPPIProp> = ({
+  timeseries,
+  timeseries_callout,
+  last_updated,
+}) => {
+  const { t, i18n } = useTranslation(["dashboard-services-producer-prices", "agencies", "common"]);
+
+  const INDEX_OPTIONS: Array<OptionType> = [
+    { label: t("keys.growth_yoy"), value: "growth_yoy" },
+    { label: t("keys.growth_qoq"), value: "growth_qoq" },
+    { label: t("keys.index"), value: "index" },
+  ];
+  const SHADE_OPTIONS: Array<OptionType> = [
+    { label: t("keys.no_shade"), value: "no_shade" },
+    { label: t("keys.recession"), value: "recession" },
+  ];
+
+  const { data, setData } = useData({
+    minmax: [0, timeseries.data[INDEX_OPTIONS[0].value].x.length - 1],
+    index: INDEX_OPTIONS[0].value,
+    shade: SHADE_OPTIONS[0].value,
+  });
+  const LATEST_TIMESTAMP = timeseries.data[data.index].x[timeseries.data[data.index].x.length - 1];
+  const { coordinate } = useSlice(timeseries.data[data.index], data.minmax);
+
+  const shader = useCallback<(key: string) => ChartDataset<keyof ChartTypeRegistry, any[]>>(
+    (key: string) => {
+      if (key === "no_shade")
+        return {
+          data: [],
+        };
+
+      return {
+        type: "line",
+        data: coordinate[key],
+        backgroundColor: AKSARA_COLOR.BLACK_H,
+        borderWidth: 0,
+        fill: true,
+        yAxisID: "y2",
+        stepped: true,
+      };
+    },
+    [data]
+  );
+
+  const getChartData = (charts: string[]): TimeseriesChartData[] =>
+    charts.map(name => {
+      const isPercentage: boolean = ["growth_qoq", "growth_yoy"].includes(data.index);
+      return {
+        title: t(`keys.${name}`),
+        prefix: isPercentage ? "" : "RM ",
+        label: t(`keys.${name}`),
+        data: coordinate[name],
+        fill: data.shade === "no_shade",
+        stats: [
+          {
+            title: t("common:common.latest", {
+              date: toDate(LATEST_TIMESTAMP, "qQ yyyy", i18n.language),
+            }),
+            value: [
+              isPercentage ? "" : "RM ",
+              numFormat(
+                Math.abs(timeseries_callout.data[data.index][name]?.latest),
+                "compact",
+                isPercentage ? 1 : 0,
+                "short",
+                i18n.language,
+                false
+              ),
+              isPercentage ? "%" : ``,
+            ].join(""),
+          },
+        ],
+      };
+    });
+
+  const detailsChartData = getChartData(
+    ServicesPPIData.filter(data => data !== "x" && data !== "total" && data !== "recession")
+  );
```

Comment:
You can try returning the `<Timeseries />` itself instead of the json. Something like this:
```ts
 const plotTimeseries = (charts: Exclude<DemographyKeys, "x">[], play: boolean) => {
    return charts.map(name => {
      const {
        title,
        label,
        data: datum,
        stats,
      } = {
        title: t(`keys.${name}`),
        label: t(`keys.${name}`),
        data: coordinate[name],
        stats: [
          {
            title: t("common:time.daily"),
            value: `${demography_callout.data[name].daily > 0 ? "+" : ""}${numFormat(
              demography_callout.data[name].daily,
              "standard"
            )}`,
          },
          {
            title: t("common:time.yearly"),
            value: numFormat(demography_callout.data[name].yearly, "standard"),
          },
        ],
      };

      return (
        <Timeseries
          key={title}
          title={title}
          className="h-[300px]"
          isLoading={data.loading}
          enableAnimation={!play}
          enableMajorTick
          displayType="standard"
          precision={[1, 0]}
          interval={config.period}
          data={{
            labels: coordinate.x,
            datasets: [
              {
                type: "line",
                label,
                data: datum,
                backgroundColor: AKSARA_COLOR.PURPLE_H,
                borderColor: AKSARA_COLOR.PURPLE,
                fill: true,
                borderWidth: 1.5,
              },
            ],
          }}
          stats={stats}
        />
      );
    });
  };
```

Then can just call the fn inside jsx eg.

```ts
      <SliderProvider>
        {play => (
          <div className="space-y-8">
            <div className="grid grid-cols-1 gap-8 lg:grid-cols-2">
              {plotTimeseries(["male", "female"], play)}
            </div>

            <Slider
              className="pt-0"
              type="range"
              period={config.period}
              value={data.minmax}
              data={demography.data[config.key].x}
              onChange={e => setData("minmax", e)}
            />

            <div className="grid grid-cols-1 gap-8 lg:grid-cols-3">
              {plotTimeseries(
                ["baby", "child", "teenager", "adult_young", "adult", "elderly"],
                play
              )}
            </div>
          </div>
        )}
      </SliderProvider>
```